### PR TITLE
Return const reference types instead of value types from SoAs

### PR DESCRIPTION
**Pull request by sslattery**

### CODE REVIEW

File: core/src/Cabana_Tuple.hpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ get( Tuple_t& tp )
 
 // Rank-0 const
 template <std::size_t M, class Tuple_t>
-KOKKOS_FORCEINLINE_FUNCTION typename Tuple_t::template member_value_type<M>
+KOKKOS_FORCEINLINE_FUNCTION typename Tuple_t::template member_const_reference_type<M>
```
Comment (dalg24) [inline core/src/Cabana_Tuple.hpp:62]:
```suggestion
KOKKOS_FORCEINLINE_FUNCTION decltype(auto)
```

**Review by streeve — APPROVED** [review]

**Comment by sslattery** [comment]
format this please

File: core/src/Cabana_Tuple.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/Cabana_Tuple.hpp]: ping...

**Comment by dalg24** [comment]
Also I thought we wanted to avoid format only commits

**Pull request merged**